### ISSUE TITLE
Actions should report invalid workflow file

### ISSUE THREAD

**Issue by ChristianSch** [go-gitea/gitea#23275]
### Feature Description

I think it would be very helpful to get an indication in case gitea detected a workflow file but deemed it invalid. Take this workflow file `main.yml`:

```
name: Gitea Actions Demo
run-name: ${{ github.actor }} is testing out Gitea Actions 🚀
on: [push]
jobs:
  Explore-Gitea-Actions:
    runs-on: ubuntu-latest
    steps:
      - run: echo "🎉 The job was automatically triggered by a ${{ github.event_name }} event."
      - run: echo "GITHUB_API_URL: ${{ GITHUB_API_URL }} GITHUB_SERVER_URL: ${{ GITHUB_SERVER_URL }} github.api_url: ${{ github.api_url }} github.server_url: ${{ github.server_url}}""
      - run: echo "done"
```

The logs report the following:
```
2023-03-03 22:38:02 2023/03/03 21:38:02 ...actions/workflows.go:69:DetectWorkflows() [W] ignore invalid workflow "main.yml": yaml: line 9: mapping values are not allowed in this context
```

But no action is started and no feedback is given to the user. I feel like this belongs in the UI. 

PS: The problem with this is the `:` which aren't allowed.

**Comment by techknowlogick** [comment]
> PS: The problem with this is the : which aren't allowed.

I think it might be due to the `""` at the end of line 9.

Regardless, I agree that reporting on invalid files should occur.

**Comment by ChristianSch** [comment]
I was actually thinking about reporting that as well, I actually got a run with two closing `"`, but thought it was such a small little detail to not report it. I might as well try to reproduce it, I can't right now. Seems (also as per https://www.yamllint.com) that `:` inside a string are not valid 🤷 

This doesn't run as well:
```
name: Gitea Actions Demo
run-name: ${{ github.actor }} is testing out Gitea Actions 🚀
on: [push]
jobs:
  Explore-Gitea-Actions:
    runs-on: ubuntu-latest
    steps:
      - run: echo "GITHUB_API_URL: ${{ GITHUB_API_URL }} GITHUB_SERVER_URL: ${{ GITHUB_SERVER_URL }} github.api_url: ${{ github.api_url }} github.server_url: ${{ github.server_url}}"
```

**Comment by techknowlogick** [comment]
>  Edit 2: yamllint.com thinks closing with "" is fine as well 😖

oh dear, thanks for those additional details.

**Comment by wxiaoguang** [comment]
See more details in #23277 (i think they are the same problem)

* https://github.com/go-gitea/gitea/issues/23277#issuecomment-1454351341

The problem is that the step of invalid shell command is wrongly reported as succeeded.

I haven't looked into details and have no idea about why the step succeeds.

**Comment by ChristianSch** [comment]
They are a bit different though: in this ticket no action is started and you don't get *any* feedback in the UI and the error is within gitea I guess. #23277 happens in the runner and seems to ignore the return code or something. In the latter ticket, an action run is started and you see something in the UI.

**Comment by wxiaoguang** [comment]
Sorry for my misunderstanding. I see your point. Yup, it's better to report the invalid workflow file to the Web UI.

**Comment by ChristianSch** [comment]
Thank you for the feedback, I clarified it in the original comment 👍

**Comment by wolfogre** [comment]
Closed by #23404
